src/lang/std/artifactGraph.ts

```diff
@@ -0,0 +1,645 @@
+import { PathToNode, Program, SourceRange } from 'lang/wasm'
+import { Models } from '@kittycad/lib'
+import { getNodePathFromSourceRange } from 'lang/queryAst'
+import { err } from 'lib/trap'
+
+interface CommonCommandProperties {
+  range: SourceRange
+  pathToNode: PathToNode
+}
+
+export interface _PlaneArtifact {
+  type: 'plane'
+  pathIds: Array<string>
+  codeRef: CommonCommandProperties
+}
+export interface PlaneArtifact {
+  type: 'plane'
+  paths: Array<_PathArtifact>
+  codeRef: CommonCommandProperties
+}
+
+export interface _PathArtifact {
+  type: 'path'
+  planeId: string
+  segIds: Array<string>
+  extrusionId: string
+  solid2dId?: string
+  codeRef: CommonCommandProperties
+}
+
+interface _solid2D {
+  type: 'solid2D'
+  pathId: string
+}
+export interface PathArtifact {
+  type: 'path'
+  plane: _PlaneArtifact | _WallArtifact
+  segs: Array<_SegmentArtifact>
+  extrusion: _ExtrusionArtifact
+  codeRef: CommonCommandProperties
+}
+
+interface _SegmentArtifact {
+  type: 'segment'
+  pathId: string
+  surfId: string
+  edgeIds: Array<string>
+  edgeCutId?: string
+  codeRef: CommonCommandProperties
+}
+interface SegmentArtifact {
+  type: 'segment'
+  path: _PathArtifact
+  surf: _WallArtifact
+  edges: Array<_ExtrudeEdge>
+  edgeCut?: _EdgeCut
+  codeRef: CommonCommandProperties
+}
+
+interface _ExtrusionArtifact {
+  type: 'extrusion'
+  pathId: string
+  surfIds: Array<string>
+  edgeIds: Array<string>
+  codeRef: CommonCommandProperties
+}
+interface ExtrusionArtifact {
+  type: 'extrusion'
+  pathId: string
+  surfs: Array<_WallArtifact | _CapArtifact>
+  edges: Array<string>
+  codeRef: CommonCommandProperties
+}
+
+interface _WallArtifact {
+  type: 'wall'
+  segId: string
+  edgeCutEdgeIds: Array<string>
+  extrusionId: string
+  pathIds: Array<string>
+}
+interface _CapArtifact {
+  type: 'cap'
+  subType: 'start' | 'end'
+  edgeCutEdgeIds: Array<string>
+  extrusionId: string
+  pathIds: Array<string>
+}
+
+interface _ExtrudeEdge {
+  type: 'extrudeEdge'
+  segId: string
+  extrusionId: string
+  edgeId: string
+}
+
+/** A edgeCut is a more generic term for both fillet or chamfer */
+interface _EdgeCut {
+  type: 'edgeCut'
+  subType: 'fillet' | 'chamfer'
+  consumedEdgeId: string
+  edgeIds: Array<string>
+  surfId: string
+  codeRef: CommonCommandProperties
+}
+
+interface EdgeCutEdge {
+  type: 'edgeCutEdge'
+  edgeCutId: string
+  surfId: string
+}
+
+export type Artifact =
+  | _PlaneArtifact
+  | _PathArtifact
+  | _SegmentArtifact
+  | _ExtrusionArtifact
+  | _WallArtifact
+  | _CapArtifact
+  | _ExtrudeEdge
+  | _EdgeCut
+  | EdgeCutEdge
```

Comment:
Leading underscore means private?

Reply:
I believe so, not 100% sure here

Reply:
`EdgeCutEdge` doesn't appear to be exported or different from the other interfaces, so it seems to match them in terms of how "private" they are at least to me

Reply:
Sorry this is something I did quickly and really needed to come back to the names.

The `_` version is the stub version, has strings ids, and the non `_` has it's properties with the actually next nodes (but the stubbed versions so that it's not linking forever)
![image](https://github.com/user-attachments/assets/de5f906f-ddda-44d1-ab0e-2eaa6dba73a2)

`_` was never going to stay sorry because really that's the default type I might go with
![image](https://github.com/user-attachments/assets/42ee206b-c1fb-4fa2-992e-76515f037248)

Reply:
Thanks for reviewing my shit, I've just been looking at this code too much.